重新設計 Branding 頁面

pan93412 opened the pull request:
三年前 (f670d5717edb5f7d6074344959b02862e0b9d59c) 設計了一版有點醜的版本：

<img width="2976" height="1598" alt="CleanShot 2025-07-23 at 19 42 12@2x" src="https://github.com/user-attachments/assets/8debe5c5-a984-455c-a0ed-bd64ed7a336b" />

最近覺得這個頁面真的太醜了，加上也更熟悉相關的設計工具了，所以用 [Figma](https://www.figma.com/design/BFRDLGasojfkEjnX21Dy5J/SITCON-Branding) 重新設計了一個新的版本，然後交由 Gemini 和 Claude 落地：

<img width="3232" height="2000" alt="CleanShot 2025-07-23 at 19 43 28@2x" src="https://github.com/user-attachments/assets/e4a0cc3a-d3d9-4e4a-b23f-ea859c39d972" />

另外也用了 BEM 來規範 CSS class name 防止衝突。

結果參考（也有 [Live Preview](https://sitcon-branding-dev-redesign.zeabur.app/)）：

<details>

<summary>macOS, Chrome</summary>

<img width="2880" height="1834" alt="Desktop zh-TW" src="https://github.com/user-attachments/assets/ebe5c03e-ebdb-4da9-a428-0ba361aedab9" />
<img width="2880" height="1960" alt="Desktop en-US" src="https://github.com/user-attachments/assets/a5301285-86f9-4991-b688-79785b8ccdc1" />

</details>

<details>

<summary>macOS, Chrome (Mobile, 425px)</summary>

<img width="850" height="868" alt="Mobile en-US" src="https://github.com/user-attachments/assets/d7160b5a-df23-4514-8980-0cc63664ca2c" />
<img width="850" height="868" alt="Mobile zh-TW" src="https://github.com/user-attachments/assets/dca7f023-31d7-44c7-a057-638943032d76" />

</details>

<details>

<summary>macOS, Safari</summary>

<img width="2770" height="2002" alt="Desktop zh-TW" src="https://github.com/user-attachments/assets/9fa1b3b5-1d9d-4992-867e-c521a19bd337" />
<img width="2770" height="2002" alt="Desktop en-US" src="https://github.com/user-attachments/assets/bfdd38de-930b-4106-afd8-d62d69296773" />

</details>

<details>

<summary>Windows, Edge</summary>

<img width="3248" height="2016" alt="Desktop zh-TW" src="https://github.com/user-attachments/assets/310f654e-f36f-4218-be44-d4eae08e209f" />
<img width="3248" height="2016" alt="Desktop en-US" src="https://github.com/user-attachments/assets/09500e89-1109-4256-819a-68f54d9c8459" />

</details>

<details>

<summary>Fedora, Firefox</summary>

  
<img width="3024" height="1890" alt="Desktop zh-TW" src="https://github.com/user-attachments/assets/a3ed48aa-17bc-40a5-9fc1-def91b23337c" />
<img width="3024" height="1890" alt="Desktop en-US" src="https://github.com/user-attachments/assets/e05fa886-3999-4667-a544-744cc42b9606" />

</details>

pan93412 committed:
Redesign branding page

pan93412 committed:
Update English translation and style

pan93412 committed:
Remove unused layout CSS

pan93412 committed:
Add licenses for icon

rschiang commented:
感謝貢獻！不過圖示授權可能不適合放在 PR 裡。(・ω・)

（另外 PR 中的 ISC License 也與 [OSI 標準](https://opensource.org/license/isc-license-txt)不合，請問這份授權文字的來源是？）

pan93412 commented:
> 感謝貢獻！不過圖示授權可能不適合放在 PR 裡。(・ω・)
> 
> （另外 PR 中的 ISC License 也與 [OSI 標準](https://opensource.org/license/isc-license-txt)不合，請問這份授權文字的來源是？）

這裡 https://lucide.dev/license

rschiang commented:
了解，所以這是針對這項 PR 額外引入的圖示集 Lucida 的 LICENSE，那的確要跟著圖示放在一起。

pan93412 commented on assets/typography-zh.css:
```diff
@@ -0,0 +1,12 @@
+/* ==========================================================================
+   Typography Styles for English
```
```suggestion
   Typography Styles for Chinese
```

pan93412 committed:
Fix typo in typography-zh.css

rschiang commented on assets/typography.css:
```diff
@@ -1,18 +1,223 @@
+   ========================================================================== */
+
 html {
-    font-family: 'Nimbus Sans', Helvetica, Arial, 'Noto Sans TC', sans-serif;
```
[Nimbus Sans](https://en.wikipedia.org/wiki/Nimbus_Sans#Nimbus_Sans_L) 是 logo branding 的一部分，所以這部分應該要留著

pan93412 replied:
<img width="3216" height="1982" alt="CleanShot 2025-07-23 at 23 12 19@2x" src="https://github.com/user-attachments/assets/ac783f12-9869-4857-8393-1b2f8e525a3d" />

是這樣說沒錯，但這個文字的 baseline 設計似乎有點小狀況
我是用 Adobe Fonts 的版本，有設定適當的 `line-height` 了

或許還是讓這個字體留在 Illustrator 上就好？

pan93412 replied:
<img width="3216" height="1982" alt="CleanShot 2025-07-23 at 23 14 13@2x" src="https://github.com/user-attachments/assets/7a345c95-0554-4300-bf2c-8f27bd91dc03" />

不過 Nimbus Sans L 就會比較正常，我用這個版本好了

rschiang commented on assets/typography.css:
```diff
@@ -1,18 +1,223 @@
+
 html {
-    font-family: 'Nimbus Sans', Helvetica, Arial, 'Noto Sans TC', sans-serif;
+    font-family: 'PingFang TC', '微軟正黑體', 'Microsoft JhengHei', 'Noto Sans TC', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, sans-serif;
```
這邊應該不需要用  `-apple-system, BlinkMacSystemFont` 了，可以用 `system-ui`（已經 [baseline](https://caniuse.com/?search=system-ui) 了）取代。

rschiang commented:
因為看起來還在 working，我先把這項 PR 調整成草稿，這樣專案成員就不會一直收到通知信XD

pan93412 committed:
Refactor Typography structure to Nimbus Sans

pan93412 committed:
Set line-height to 1 for nav-link

pan93412 committed:
Make colors apply on icons

pan93412 committed:
Use Nimbus Sans L

pan93412 committed:
Apply English fonts on Chinese typography

pan93412 committed:
Set the title and button text to a line height of 1

pan93412 commented:
我這裡也補上其他瀏覽器的顯示情況了～

pan93412 committed:
Apply icons changes to english.html

pan93412 committed:
Set the "line-height" of the example title to 1

rschiang committed:
Refactor: Merge multiple typography stylesheets into one

This commit tidies up the typography stylesheets by merging the
files and utilizing `:lang` selector for additional per-language
fine tuning.

The font fallback stack is also heavily simplified in the following
logic:

* `system-ui` is widely available now -- there is no need to use
  `BlinkMacSystemFont` or `-apple-system` anymore.
* Source Han Sans TC (a.k.a. Noto Sans TC) and Nimbus Sans L are
  part of SITCON branding and shall not be treated as fallback
  fonts.
* We're not designing a web app -- only utilize `system-ui` to
  avoid falling back to uncontrollable `sans-serif`.

rschiang committed:
Refactor: remove redundant definitions & extract variables

rschiang reviewed (changes requested):
我把 PR 看過了！覺得大方向沒什麼問題，但 CSS 樣式表感覺非常疊床架屋、而且有很多重複複寫之處，讓我有點困惑（是不是使用 AI 輔助後沒有清理乾淨？）：

* 舉例來說，原本的程式碼在 `:root` 定義了大量的間距、字型大小與 breakpoint，卻又沒有統一使用這些定義的變數、甚至用了不在變數裡面的值；
* 在 `layout.css` 定義的字體大小與顏色在 `typography.css` 複寫一輪、而且兩邊級距不一樣；
* `typography.css` 有點難以理解的分成三個檔案，而且其實只有中文的一項選擇器需要特殊處理。這在事後會經過編譯的 SCSS 底下可以理解，不過在 vanilla CSS 等於要送多次 request 才能 render 網頁，其實會效能不彰；
* `.examples__icon` 的三個子類別重複定義了 `width` 與 `height`、`@media (min-width: 48rem)` 重複定義了一模一樣的 `font-size`；
* 混用 `rem` 與 `px`（理論上來說，如果要支援無障礙縮放，應該統一使用 `rem` 定義尺寸）
* 在定義全域樣式時混用 `:root`、`html`、`body`，使得樣式表繼承階層不明確

⋯⋯等等。

我這邊先做了些初步的清理；不知道能不能請 @pan93412 重新檢視一下看看變數跟尺寸怎麼統合？整理乾淨後應該就可以 merge。

pan93412 committed:
Refactor: unify CSS variable definitions, remove duplicate and inconsistent variables

pan93412 committed:
Refactor: use rem units consistently, remove mixed px usage

pan93412 committed:
Refactor: merge CSS files to one file

pan93412 committed:
Fix the text color when hovering on button

pan93412 committed:
Refactor: merge duplicate media queries

pan93412 committed:
Refactor: clear out magic numbers

pan93412 committed:
Reformat with prettier

pan93412 committed:
Fix description padding

pan93412 committed:
Refactor: rename style to style.css

pan93412 committed:
Eliminate non-maintainable elements

pan93412 commented:
@rschiang 我這裡把 code 清完了，不過看起來 commits 數有點太多了，可能要用 squash merge 的方式併進去了

Online Preview: https://sitcon-branding-dev-redesign.zeabur.app/
樣式經過標準化後有些微差異，但應該不會差太多。

pan93412 commented:
ping @rschiang

rschiang reviewed (approved):
（因為出遠門去爬山晚了點回覆，不好意思）

我覺得沒有什麼大問題，之前提到的一致性問題看起來都有修正了！

rschiang commented:
PR 已合併！因為覺得這些討論紀錄還滿珍貴的，我就不 squash 了。(・ω・)ノ
